llvm/lib/Transforms/Utils/BasicBlockUtils.cpp

```diff
@@ -333,6 +333,9 @@ bool llvm::MergeBlockIntoPredecessor(BasicBlock *BB, DomTreeUpdater *DTU,
   // Finally, erase the old block and update dominator info.
   DeleteDeadBlock(BB, DTU);
 
+  // Remove redundant "llvm.dbg" instrunctions after blocks merged.
+  RemoveRedundantDbgInstrs(PredBB);
```

Comment:
This needs to do a forward and backward scan over the BB, even if there are no debug intrinsics in the block. Can it only be called when we cloned blocks with debug intrinsics?

Reply:
Thanks a lot, @fhahn , I will check it

Reply:
@fhahn, I added a condition to control it (only execute with the "-g" flag ). Could you please review it again?